app/src/main/java/com/hedvig/app/feature/embark/EmbarkViewModel.kt

```diff
@@ -158,8 +158,12 @@ abstract class EmbarkViewModel(
                 //  meaning that the old values were returned from getList/get.
                 valueStore.withCommittedVersion {
                     val quoteCartId = this.get(QUOTE_CART_ID_KEY)?.let { QuoteCartId(it) }
-                    val selectedContractTypes = nextPassage.getSelectedContractTypes(this)
-                    _events.trySend(Event.Offer(quoteCartId, selectedContractTypes))
+                    if (quoteCartId == null) {
+                        _events.trySend(Event.Error())
+                    } else {
```

Comment:
If we've reached the point where `key != null` aka `nextPassage?.getOfferKeyOrNull(valueStore)` returned a non-null result, then we must have a QuoteCartId in the store. If not, as I understand it there's a big problem anyway, therefore we can error here. Would love to know if this understanding of mine is wrong.